### PR TITLE
Drafts

### PR DESCRIPTION
**Pull request by belucid**
https://trello.com/c/4ptAIHYc

Back end changes that repurpose old story draft support for entry drafts. 

Supports the Web UI work here: https://github.com/open-company/open-company-web/pull/282

- [x] Review the code
- [x] Follow the tests steps on https://github.com/open-company/open-company-web/pull/282

Be sure to have a 2nd user logged into the org during your tests and watch when they get change notifications:

- [x] No change when the other user creates a draft?
- [x] No change when the other user updates a draft?
- [x] No change when the other user deletes a draft?
- [x] Do get a change when the other user publishes a draft?
- [x] Still get a change when the other user goes straight to a published entry with no draft (no regression)?
- [x] Still get a change when the other user updates a published entry (no regression)?
- [x] Still get a change when the other user deletes a published entry (no regression)?

### CODE REVIEW

File: src/oc/storage/api/entries.clj [hunk 1]
```diff
@@ -267,15 +285,73 @@
...
+                            :post (api-common/only-accept 406 mt/entry-media-type)})
+
+  ;; Possibly no data to handle
+  :malformed? (by-method {
+    :options false
+    :post (fn [ctx] (api-common/malformed-json? ctx true))}) ; allow nil
+  :processable? (by-method {
+    :options true
+    :post (fn [ctx] (let [entry-props (:data ctx)]
+                      (or (nil? entry-props) ; no share is fine
```
Comment (bago2k4) [inline src/oc/storage/api/entries.clj]: probably this comment came from the old API where you could share a story while publishing it.

**Comment by bago2k4** [comment]
@belucid i'm not done yet, i just reviewed the code.

**Comment by belucid** [comment]
@bago2k4 I removed old instances of `story` and `stories` from the code. The very few remaining instances are intentional (such as the migrations).

File: src/oc/storage/api/entries.clj (hunk 1 above)
Comment (belucid) [reply src/oc/storage/api/entries.clj]: agree on the comment, it's updated

**Pull request merged**